lib/devices/vacuum.js

```diff
@@ -9,7 +9,10 @@ const MiioApi = require('../device');
 const BatteryLevel = require('./capabilities/battery-level');
 
 function checkResult(r) {
-	if(r !== 0) {
+	//console.log(r)
+	// {"result":0,"id":17} 	  = Firmware 3.3.9_003095 (Gen1)
+	// {"result":["ok"],"id":11}  = Firmware 3.3.9_003194 (Gen1), 3.3.9_001168 (Gen2)
+	if( r !== 0 && r[0] !== 'ok' ) {
```

Comment:
@nicoh88 might want to check if r is an array plus add null check to make it robust?